src/main/java/seedu/exercise/logic/commands/events/EventFactory.java

```diff
@@ -21,31 +25,86 @@
      * Generates an Event object that can execute the behaviour of a given Command as well
      * as its opposite behaviour.
      *
-     * @param command a {@code Command} to be represented with using an Event object
+     * @param undoableCommand a {@code UndoableCommand} to be represented with using an Event object
      * @return an {@code Event} that can be undone or redone
+     * @throws CommandException if command provided is not undoable
      */
-    static Event commandToEvent(UndoableCommand command) throws CommandException {
-        if (command instanceof AddExerciseCommand) {
-            Exercise exercise = ((AddExerciseCommand) command).getExercise();
-            return new AddExerciseEvent(exercise);
-
-        } else if (command instanceof DeleteExerciseCommand) {
-            Exercise exercise = ((DeleteExerciseCommand) command).getExercise();
-            return new DeleteExerciseEvent(exercise);
-
-        } else if (command instanceof EditCommand) {
-            Exercise exerciseOld = ((EditCommand) command).getExerciseToEdit();
-            Exercise exerciseNew = ((EditCommand) command).getEditedExercise();
-            return new EditEvent(exerciseOld, exerciseNew);
-
-        } else if (command instanceof ClearCommand) {
-            ReadOnlyResourceBook<Exercise> exerciseBookCleared =
-                new ReadOnlyResourceBook<>(((ClearCommand) command).getExerciseBookCleared());
-            return new ClearEvent(exerciseBookCleared);
-
-        } else {
+    static Event commandToEvent(UndoableCommand undoableCommand) throws CommandException {
+        Command command = (Command) undoableCommand;
+        String commandWord = command.getCommandWord();
+
+        switch (commandWord) {
+        case AddCommand.COMMAND_WORD:
+            return addCommandToEvent((AddCommand) command);
+
+        case DeleteCommand.COMMAND_WORD:
+            return deleteCommandToEvent((DeleteCommand) command);
+
+        case EditCommand.COMMAND_WORD:
+            return new EditEvent(((EditCommand) command).getEventPayload());
+
+        case ClearCommand.COMMAND_WORD:
+            return new ClearEvent(((ClearCommand) command).getEventPayload());
+
+        default:
+            throw new CommandException(
```

Comment:
There are multiple copies of the method `getEventPayload()`. I think it's better to have a contract for this method so you can have Commands that have some payload implement an interface like `PayloadableCommand`. Think of a better name for this but you get the idea

Reply:
Added a `PayloadCarrierCommand` with a `getPayload()` method